client/signup/jetpack-connect/site-url-input.jsx

```diff
@@ -15,21 +15,26 @@ import FormTextInput from 'components/forms/form-text-input';
 import Spinner from 'components/spinner';
 import untrailingslashit from 'lib/route/untrailingslashit';
 
-
 class JetpackConnectSiteURLInput extends Component {
-	constructor() {
-		super();
-		this.onChange = this.onChange.bind( this );
-		this.onClick = this.onClick.bind( this );
-		this.handleKeyPress = this.handleKeyPress.bind( this );
-	}
-
-	componentWillMount() {
-		if ( this.props.url ) {
-			this.setState( { value: untrailingslashit( this.props.url ), shownValue: this.props.url } );
-		} else {
-			this.setState( { value: '', shownValue: '' } );
-		}
+	static propTypes = {
+		handleOnClickTos: PropTypes.func,
+		isError: PropTypes.oneOfType( [
+			PropTypes.string,
+			PropTypes.bool,
+		] ),
+		isFetching: PropTypes.bool,
+		isInstall: PropTypes.bool,
+		onChange: PropTypes.func,
+		onClick: PropTypes.func,
+		url: PropTypes.string,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = this.props.url
```

Comment:
You're using `this.props` anyway, so If we move `this.state` to `state = ...` we'll no longer need to declare a `constructor`, right?